test/spec/common/client/set-cache.js

```diff
@@ -0,0 +1,22 @@
+'use strict';
+
+var test = require('tape');
+
+test('cache is not empty object when specified in opts', function(t) {
+  t.plan(1);
+
+  var opts = {
+    cache: {
+      'path/for/query': {}
+    }
+  };
+
+  var algoliasearch = require('../../../../');
+  var search = algoliasearch('applicationID', 'apiKey', opts);
+
+  t.deepEqual(
+    search.cache,
```

Comment:
This will decouple the test from the internal implementation that might not store the cache inside search.cache in the future, but maybe in another property/hidden.